azext_iot/iothub/params.py

```diff
@@ -246,3 +248,73 @@ def load_iothub_arguments(self, _):
             options_list=["--content-type", "--ct"],
             help="MIME Type of file.",
         )
+
+    with self.argument_context("iot hub state") as context:
+        context.argument(
+            "state_file",
+            options_list=["--state-file", "-f"],
+            help="The path to the file where the state information will be stored."
+        )
+        context.argument(
+            "replace",
+            options_list=["--replace", "-r"],
+            help="If this flag is set, then the command will delete the current devices, configurations, and certificates "
+                 "of the destination hub."
+        )
+        context.argument(
+            "hub_aspects",
+            options_list=["--aspects"],
+            nargs="*",
+            arg_type=get_enum_type(HubAspects),
+            help="Hub Aspects (space seperated)."
+        )
+
+    with self.argument_context("iot hub state import") as context:
+        context.argument(
+            "replace",
+            options_list=["--replace", "-r"],
+            help="If this flag is set, then the command will overwrite the contents of the output file."
+        )
+
+    with self.argument_context("iot hub state migrate") as context:
+        context.argument(
+            "hub_name",
+            options_list=["--destination-hub", "--dh"],
+            help="Name of IoT Hub to which the origin hub will be copied."
+        )
+        context.argument(
+            "resource_group_name",
+            options_list=["--destination-resource-group", "--dg"],
+            help="Name of resource group of the IoT Hub to which the origin hub will be copied."
+        )
+        context.argument(
+            "login",
+            options_list=["--destination-hub-login", "--dl"],
+            validator=mode2_iot_login_handler,
+            help="This command supports an entity connection string with rights to perform action on the destination hub. "
```

Comment:
Here and for the orig_hub_login, if a connection string is provided, then theoretically only dataplane should be serviced.

Reply:
would like to debate about this a bit more - if a user provides any dataplane oriented params - should we lock them to only dataplane? what if they do dataplane + aspects arm?

Reply:
In the IoT CLI the point of providing a connection string has been for data plane only. We also advertise usage of the connection string to avoid having to 'az login', as in when a connection string is provided, your AAD credentials will not be utilized meaning one does _need_ to be logged in.

In regards to your comments 

> if a user provides any dataplane oriented params - should we lock them to only dataplane?

If the user provides a connection string, then yes they should be locked to data plane. Only the connection string policy permissions are in play for what a user can & can't do.

>what if they do dataplane + aspects arm?

If the user provides a connection string, then tries to run ARM operations, the CLI should error explaining ARM is not possible with connection string and to leverage the non-connection string path.